doc/book/admin/access_control.rst

```diff
@@ -1,377 +1,1069 @@
-In either case, there are up to five parameters:
+Privileges
+~~~~~~~~~~
 
-.. code-block:: lua
+The privileges granted to a user determine which operations the user can perform, for example:
 
-    (user-name, privilege, object-type [, object-name [, options]])
+*   The ``read`` and ``write`` privileges granted to the ``space`` :ref:`object <access_control_concepts_objects>` allow a user to read or modify data in the specified space.
+*   The ``create`` privilege granted to the ``space`` object allows a user to create new spaces.
+*   The ``execute`` privilege granted to the ``function`` object allows a user to execute the specified function.
+*   The ``session`` privilege granted to a user allows connecting to an instance over IPROTO.
 
-* ``user-name`` is the user (or role) that will receive or lose the privilege;
-* ``privilege`` is any of 'read', 'write', 'execute', 'create', 'alter', 'drop',
-  'usage', or 'session' (or a comma-separated list);
-* ``object-type`` is any of 'space', 'index',
-  'sequence', 'function', 'user', 'role', or 'universe';
-* ``object-name`` is what the privilege is for
-  (omitted if ``object-type`` is 'universe')
-  (may be omitted or ``nil`` if the intent is to grant for all objects of the same type);
-* ``options`` is a list inside braces, for example ``{if_not_exists=true|false}``
-  (usually omitted because the default is acceptable).
+Note that some privileges might require read and write access to certain system spaces.
+For example, the ``create`` privilege granted to the ``space`` object requires ``read`` and ``write`` privileges to the :ref:`_space <box_space-space>` system space.
+Similarly, granting the ability to create functions requires ``read`` and ``write`` access to the :ref:`_func <box_space-func>` space.
 
-  All updates of user privileges are reflected immediately in the existing sessions
-  and objects, e.g. functions.
+..  NOTE::
 
-**Example for granting many privileges at once**
+    Information about privileges is stored in the :ref:`_priv <box_space-priv>` space.
 
-In this example an 'admin' user grants many privileges on
-many objects to user 'U', using a single request.
 
-.. code-block:: lua
 
-    box.schema.user.grant('U','read,write,execute,create,drop','universe')
+.. _access_control_concepts_roles:
 
-.. _authentication-owners_privileges-examples-specific:
+Roles
+~~~~~
+A role is a container for :ref:`privileges <authentication-owners_privileges>` that can be granted to users.
+Roles can also be assigned to other roles, creating a role hierarchy.
 
-**Examples for granting privileges for specific operations**
+There are the following built-in roles in Tarantool:
 
-In these examples an administrator grants strictly
-the minimal privileges necessary for particular operations,
-to user 'U'.
+*   ``super`` has all available administrative permissions.
+*   ``public`` is automatically granted to new users when they are created.
```

Comment:
Can we say anything specific or give a usage advice about this role? The current definition gives me no idea why is it needed and how to use it...